packages/ubuntu_desktop_installer/lib/pages/installation_slides/installation_slides_page.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:subiquity_client/subiquity_client.dart';
+
+import '../../constants.dart';
+import '../wizard_page.dart';
+import 'installation_slides_model.dart';
+
+class InstallationSlidesPage extends StatefulWidget {
+  const InstallationSlidesPage({
+    Key? key,
+  }) : super(key: key);
+
+  static Widget create(BuildContext context) {
+    final client = Provider.of<SubiquityClient>(context, listen: false);
+    return ChangeNotifierProvider(
+      create: (_) => InstallationSlidesModel(client),
+      child: InstallationSlidesPage(),
+    );
+  }
+
+  @override
+  InstallationSlidesPageState createState() => InstallationSlidesPageState();
+}
+
+class InstallationSlidesPageState extends State<InstallationSlidesPage> {
+  @override
+  void initState() {
+    super.initState();
+
+    final model = Provider.of<InstallationSlidesModel>(context, listen: false);
+    model.init();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final model = Provider.of<InstallationSlidesModel>(context);
+    return WizardPage(
+      title: const Text('Welcome to Ubuntu'),
+      content: Center(
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            if (model.isInstalling) CircularProgressIndicator(),
+            const SizedBox(height: kContentSpacing),
+            Text(
+              model.isDone
+                  ? 'Installation complete.\n\nPlease restart the machine.'
+                  : model.hasError
+                      ? 'Something went wrong.\n\nPlease restart the machine.'
+                      : 'Installing... (${model.currentStep}/${model.totalSteps})',
+              textAlign: TextAlign.center,
+            ),
+          ],
+        ),
+      ),
+      actions: <WizardAction>[
+        WizardAction(
+          label: 'Restart',
+          enabled: model.isDone || model.hasError,
+          onActivated: model.reboot,
```

Comment:
As discussed elsewhere, let's close the app after requesting subiquity to reboot (especially useful in dry run mode, where rebooting doesn't actually order the machine to restart but shuts down the subiquity server).

Reply:
:+1: will fix when we get a response from subiquity

Reply:
i made it exit without waiting for a response that it never gets. we can add response handling when the client provides it

Reply:
SGTM, thanks.